backend-go/config.go

```diff
@@ -1,232 +1,104 @@
-		"NATS_PASSWORD": natsPassword,
-		"CONFIG_FILE":   configFile,
-		"CONFIG_KEY":    configKey,
-		"SERVICE_NAME":  serviceName,
-		"DATA_DIR":      dataDir,
-	}
-
-	// prepare output map
-	parsedArgs := map[string]any{
-		"DEV": dev,
-	}
 
-	// only populate parsedArgs with entries that were, indeed, given (dev is an exception)
-	flag.Visit(func(currentFlag *flag.Flag) {
-		if value, ok := flagValues[currentFlag.Name]; ok {
-			parsedArgs[currentFlag.Name] = value
+		key = strings.TrimPrefix(key, "--")
+		if strings.Contains(key, "=") {
+			// --KEY=VALUE
+			split := strings.Split(key, "=")
+			innerKey := strings.ToUpper(split[0]) // all config keys are uppercase
+			innerValue := strings.Join(split[1:], "=")
+			res[innerKey] = innerValue
+			continue
 		}
-	})
 
-	return &parsedArgs
-}
-
-// Read the environment variables and provides them as map ready to be included in the service config.
-func envConfig() *map[string]string {
-	result := make(map[string]string, len(os.Environ()))
-	for _, entry := range os.Environ() {
-		if key, value, ok := strings.Cut(entry, "="); ok {
-			result[key] = value
-		}
-		// we don't want to add empty env variables
-	}
-	return &result
-}
+		// --KEY (VALUE)
+		// convert the key to uppercase as it doesn't contain the value in this case:
+		key = strings.ToUpper(key)
 
-// Tries to read the configuration file and returns the content as untyped map.
-// Fails, if the config file is not readable or if the content is not JSON parsable.
-func fileConfig(configPath string) (*map[string]any, error) {
-	// Note that the file config is supposed to be a json config
-	jsonConfig := map[string]any{}
-	jsonConfigBytes, err := os.ReadFile(configPath)
+		hasValue := i+1 < len(os.Args) &&
+			!strings.HasPrefix(os.Args[i+1], "--")
 
-	if err != nil {
-		log.Printf("Config file %s could not be read: %s\n", configPath, err)
-		return nil, err
-	}
+		if hasValue {
+			// it's a key-value pair
+			res[key] = os.Args[i+1]
+			i++
+			continue
+		}
 
-	if err = json.Unmarshal(jsonConfigBytes, &jsonConfig); err != nil {
-		log.Printf("Config file %s could not be parsed: %s\n", configPath, err)
-		return nil, err
+		// it's a flag and not a key-value pair
+		res[key] = true
 	}
-
-	return &jsonConfig, nil
+	return res
 }
 
-// Returns the default configuration for development purposes.
-// Fails, if the process is not allowed to determine the current working directory.
-func devModeDefaultConfig() (*map[string]string, error) {
-	dataDir, err := filepath.Abs("data")
-	if err != nil {
-		return nil, err
-	}
-
-	return &map[string]string{
-		"NATS_URL":     nats.DefaultURL,
-		"SERVICE_NAME": fmt.Sprint("dev-", os.Getgid()),
-		"DATA_DIR":     dataDir,
-	}, nil
+// EnvArgsMap returns a map of all environment variables, where the key is the uppercase key
+// and the value is the value of the environment variable.
+//
+// While this does always return a string, for compatibility with MergeMaps, it is
+// typed as map[string]any.
```

Comment:
I don't think we should restrict this to just strings. It could be feasible, may that be by an amendment of the official spec or some sort of middleware passed to the function, to have non-string values. We already have these for the CLI "flags" (as opposed to key-value arguments), which when set make this config variable `true`.

Essentially, tightening this down to always be a `map[string]string` (and also making the flags a `string`) would make a promise for this to always hold true that I'm not willing to make at this time. If we see further down the road that we never used anything apart from strings, we can probably still tighten it down (I would assume that this shouldn't be a breaking change – casting a `string` to a `string` should hopefully (?) not break anything). But at this time, all I can guarantee about configs is that they have a `string` key and a JSON-serializable value.